sbi/neural_nets/flow.py

```diff
@@ -458,69 +740,75 @@ def build_zuko_maf(
-    embedding_net.eval()
-    y_numel = embedding_net(batch_y[:1]).numel()
-    if x_numel == 1:
-        warn(
-            "In one-dimensional output space, this flow is limited to Gaussians",
-            stacklevel=1,
-        )
+
+    x_numel, y_numel = get_numel(batch_x, batch_y, embedding_net)
 
     if isinstance(hidden_features, int):
         hidden_features = [hidden_features] * num_transforms
 
-    if x_numel == 1:
-        maf = zuko.flows.MAF(
-            features=x_numel,
-            context=y_numel,
-            hidden_features=hidden_features,
-            transforms=num_transforms,
+    build_nf = getattr(zuko.flows, which_nf)
+
+    if which_nf == "CNF":
+        flow_built = build_nf(
+            features=x_numel, context=y_numel, hidden_features=hidden_features, **kwargs
         )
     else:
-        maf = zuko.flows.MAF(
+        flow_built = build_nf(
             features=x_numel,
             context=y_numel,
             hidden_features=hidden_features,
             transforms=num_transforms,
-            randperm=randperm,
-            residual=residual,
+            **kwargs,
         )
 
-    transforms: Union[Sequence[LazyTransform], LazyTransform]
-    transforms = maf.transform.transforms  # pyright: ignore[reportAssignmentType]
-    z_score_x_bool, structured_x = z_score_parser(z_score_x)
-    if z_score_x_bool:
-        # transforms = transforms
-        transforms = (
-            *transforms,
-            # Ideally `standardizing_transform` would return a `LazyTransform` instead of ` AffineTransform | Unconditional`, maybe all three are compatible
-            standardizing_transform(batch_x, structured_x, backend="zuko"),  # pyright: ignore[reportAssignmentType]
-        )
+    # Continuous normalizing flows (CNF) only have one transform,
+    # so we need to handle them slightly differently.
+    if which_nf == "CNF":
+        transform = flow_built.transform
 
-    z_score_y_bool, structured_y = z_score_parser(z_score_y)
-    if z_score_y_bool:
-        # Prepend standardizing transform to y-embedding.
-        embedding_net = nn.Sequential(
-            standardizing_net(batch_y, structured_y), embedding_net
-        )
+        z_score_x_bool, structured_x = z_score_parser(z_score_x)
+        if z_score_x_bool:
+            transform = (
+                transform,
+                standardizing_transform(batch_x, structured_x, backend="zuko"),
+            )
 
-    # Combine transforms.
-    neural_net = zuko.flows.Flow(transforms, maf.base)
+        z_score_y_bool, structured_y = z_score_parser(z_score_y)
+        if z_score_y_bool:
+            # Prepend standardizing transform to y-embedding.
+            embedding_net = nn.Sequential(
+                standardizing_net(batch_y, structured_y), embedding_net
+            )
+
+        # Combine transforms.
+        neural_net = zuko.flows.Flow(transform, flow_built.base)  # pyright: ignore[reportArgumentType]
```

Comment:
Please remove all the `pyright ignore[reportArgumentType]`. #1099 has fixed this issue.
```suggestion
        neural_net = zuko.flows.Flow(transform, flow_built.base)
```